### ISSUE TITLE
IndexOutOfBoundsException from CKMSQuantiles

### ISSUE THREAD

**Issue by trustin** [micrometer-metrics/micrometer#172]
Getting this in production. Micrometer 0.10.0, but it seems like CKMSQuantiles did not change since its initial commit:

```
2017-10-24 11:06:04.129  WARN 66691 --- [armeria-common-worker-epoll-3-29] c.l.a.c.l.RequestLogListenerInvoker      : onRequestLog() failed with an exception:
java.lang.ArrayIndexOutOfBoundsException: 500
    at io.micrometer.core.instrument.stats.quantile.CKMSQuantiles.observe(CKMSQuantiles.java:81)
    at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary$Child.observe(CustomPrometheusSummary.java:132)
    at io.micrometer.core.instrument.prometheus.PrometheusTimer.record(PrometheusTimer.java:40)
    ...
```

**Comment by trustin** [comment]
Here's how we build a `CKMSQuantiles`:

```java
private static final double[] DEFAULT_QUANTILES = { 0.5, 0.75, 0.95, 0.98, 0.99, 0.999 };

final CKMSQuantiles.Builder builder = new CKMSQuantiles.Builder();
for (double q : DEFAULT_QUANTILES) {
    builder.quantile(q, 0.01);
}
return builder.create();
```

**Comment by trustin** [comment]
It seems like `CKMSQuantiles` isn't protected by a lock or a synchronized block, causing a race when called by multiple threads. Is this intentional?

**Comment by jkschneider** [comment]
Possibly connected to #139.

**Comment by jkschneider** [comment]
Fixed with 736fd95ca5a9fb9846699489877eb4021882b9df
